pkg/pool-manager/pool.go

```diff
@@ -79,7 +79,26 @@ type macEntry struct {
 	transactionTimestamp *time.Time
 }
 
-type macMap map[string]macEntry
+type macKey struct {
```

Comment:
I think this one deserves a separate file. Please move this to a new file `macKey.go`, and add the appropriate `macKey_tests.go` with unit tests for it (nothing fancy, I think this is enough:
```
var _ = Describe("MacKey", func() {
	DescribeTable("String should return the normalized MAC address",
		func(input string, expected string) {
			macKey := macKey{normalizedMacAddress: input}
			Expect(macKey.String()).To(Equal(expected))
		},
		Entry("colon-separated input", "00:00:5e:00:53:01", "00:00:5e:00:53:01"),
		Entry("hyphen-separated input", "00-00-5e-00-53-01", "00:00:5e:00:53:01"),
		Entry("period-separated input", "0000.5e00.5301", "00:00:5e:00:53:01"),
	)
})
```
)

Reply:
Nice idea! Thanks! Done